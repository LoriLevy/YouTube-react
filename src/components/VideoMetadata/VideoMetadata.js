import React from 'react';
import {Button, Divider, Icon } from 'semantic-ui-react';
import './VideoMetadata.scss';
import {Rating} from '../Rating/Rating';
export function VideoMetadata(props) {
    
    
    const viewCount = Number(props.viewCount).toLocaleString() || '';
    
    return (
    <div className="video-metadata">
        <h3>Video Title</h3>
        <div className="video-stats">
            <span>{viewCount} Views</span>
            <div className="video-actions">
                <Rating likeCount={1000}  dislikeCount={100} />
                <Button basic icon labelPosition="left">
                    <Icon name="share" />
                    Share
                </Button>

                <Button basic icon>
                    <Icon name="ellipsis horizontal" />
                </Button>
            </div>
        </div>
        <Divider />
    </div>
   );
}