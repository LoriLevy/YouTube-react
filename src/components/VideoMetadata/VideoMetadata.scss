/* src/components/VideoMetadata/VideoMetadata.scss */
@import '../../styles/shared.scss';

.video-metadata {
    color: #707070;
    font-size: 16px;
    padding-left: 15px;

    h3 {
        color: $text-color-dark;
        font-weight: 400;
    }
    .video-stats {
        display: flex;
        justify-content: space-between; /* maximize the space between children by floating the views to the left and the stats to the right */
        width: 100%;
        align-items: center;            /* vertical alignment */

    }

    .video-actions {
        display: flex;
        align-items: center;            /* vertical alignment */
            & > *:not(:last-child) {
                margin-right: 8px;
            }
    }
}
