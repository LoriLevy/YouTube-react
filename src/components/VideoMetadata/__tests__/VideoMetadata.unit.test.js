import React from 'react';
import {VideoMetadata} from '../VideoMetadata';
import {shallow} from 'enzyme';

describe('VideoMetadata', ()=> {
    test('Renders without props', () => {
        const wrapper = shallow(<VideoMetadata/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('Renders with view count', () => {
        const wrapper = shallow(<VideoMetadata viewCount={199234}/>)
        expect(wrapper).toMatchSnapshot();
    });
});