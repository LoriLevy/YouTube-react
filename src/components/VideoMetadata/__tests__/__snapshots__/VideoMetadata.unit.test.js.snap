// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoMetadata Renders with view count 1`] = `
<div
  className="video-metadata"
>
  <h3 />
  <div
    className="video-stats"
  >
    <span>
      199,234
       Views
    </span>
    <div
      className="video-actions"
    >
      <Rating
        dislikeCount="NaN"
        likeCount="NaN"
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;

exports[`VideoMetadata Renders without props 1`] = `
<div
  className="video-metadata"
>
  <h3 />
  <div
    className="video-stats"
  >
    <span>
      NaN
       Views
    </span>
    <div
      className="video-actions"
    >
      <Rating
        dislikeCount="NaN"
        likeCount="NaN"
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;
